xarray/core/groupby.py

```diff
@@ -73,6 +74,21 @@ def check_reduce_dims(reduce_dims, dimensions):
             )
 
 
+def _maybe_squeeze_indices(
+    indices, squeeze: bool | None, grouper: ResolvedGrouper, warn: bool
+):
+    if squeeze in [None, True] and grouper.can_squeeze:
+        if squeeze is None and warn:
+            emit_user_level_warning(
+                "The `squeeze` kwarg to GroupBy is being removed."
+                "Pass .groupby(..., squeeze=False) to silence this warning."
```

Comment:
I think we should say that something is happening that won't be happening; it's not a case where the only change will be the warning going away, IIUC

```suggestion
                "Pass .groupby(..., squeeze=False) to disable squeezing, and silence this warning."
```

(or maybe "disable squeezing" needs more description, not sure)